recipes/physx/4.1.1/conanfile.py

```diff
@@ -0,0 +1,266 @@
+        cmakemodules_path = os.path.join(
+            self._source_subfolder,
+            "externals",
+            "CMakeModules" if self.settings.os == "Windows" else "cmakemodules"
+        )
+        self._cmake.definitions["CMAKEMODULES_PATH"] = os.path.abspath(cmakemodules_path).replace("\\", "/")
+        self._cmake.definitions["PHYSX_ROOT_DIR"] = os.path.abspath(os.path.join(self._source_subfolder, "physx")).replace("\\", "/")
+
+        # Options defined in physx/source/compiler/cmake/CMakeLists.txt
+        if self.settings.os in ["Windows", "Android"]:
+            self._cmake.definitions["PX_SCALAR_MATH"] = not self.options.enable_simd # this value doesn't matter on other os
+        self._cmake.definitions["PX_GENERATE_STATIC_LIBRARIES"] = not self.options.shared
+        self._cmake.definitions["PX_EXPORT_LOWLEVEL_PDB"] = False
+        self._cmake.definitions["PXSHARED_PATH"] = os.path.abspath(os.path.join(self._source_subfolder, "pxshared")).replace("\\", "/")
+        self._cmake.definitions["PXSHARED_INSTALL_PREFIX"] = self.package_folder.replace("\\", "/")
+        self._cmake.definitions["PX_GENERATE_SOURCE_DISTRO"] = False
+
+        # Options defined in externals/cmakemodules/NVidiaBuildOptions.cmake
+        self._cmake.definitions["NV_APPEND_CONFIG_NAME"] = False
+        self._cmake.definitions["NV_USE_GAMEWORKS_OUTPUT_DIRS"] = False
+        if self.settings.compiler == "Visual Studio":
+            self._cmake.definitions["NV_USE_STATIC_WINCRT"] = str(self.settings.compiler.runtime).startswith("MT")
+            self._cmake.definitions["NV_USE_DEBUG_WINCRT"] = str(self.settings.compiler.runtime).endswith("d")
+        self._cmake.definitions["NV_FORCE_64BIT_SUFFIX"] = False
+        self._cmake.definitions["NV_FORCE_32BIT_SUFFIX"] = False
+        self._cmake.definitions["PX_ROOT_LIB_DIR"] = os.path.abspath(os.path.join(self.package_folder, "lib")).replace("\\", "/")
+
+        if self.settings.os == "Windows":
+            # Options defined in physx/source/compiler/cmake/windows/CMakeLists.txt
+            self._cmake.definitions["PX_COPY_EXTERNAL_DLL"] = False # External dll copy disabled, PhysXDevice dll copy is handled afterward during conan packaging
+            self._cmake.definitions["PX_FLOAT_POINT_PRECISE_MATH"] = self.options.enable_float_point_precise_math
+            self._cmake.definitions["PX_USE_NVTX"] = False # Could be controlled by an option if NVTX had a recipe, disabled for the moment
+            self._cmake.definitions["GPU_DLL_COPIED"] = True # PhysXGpu dll copy disabled, this copy is handled afterward during conan packaging
+
+            # Options used in physx/source/compiler/cmake/windows/PhysX.cmake
+            self._cmake.definitions["PX_GENERATE_GPU_PROJECTS"] = False
+
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _get_physx_build_type(self):
+        if self.settings.build_type == "Debug":
```

Comment:
dict will look more readable:
```
return {'Debug': 'debug',
             'RelWithDebInfo': 'checked',
             'Release': 'profile' if self.options.release_build_type == 'profile' else 'release'}.
             get(str(self.settings.build_type))

```

Reply:
I agree, but I was not sure if it would work when release_build_type is deleted.

Reply:
Actually, I have tested, and it fails (python error) if build_type is not Release, since options.release_build_type is deleted.

Reply:
okay, thanks for trying